pkg/rbd/controllerserver.go

```diff
@@ -87,10 +53,17 @@ func (cs *ControllerServer) validateVolumeReq(req *csi.CreateVolumeRequest) erro
 	if req.VolumeCapabilities == nil {
 		return status.Error(codes.InvalidArgument, "Volume Capabilities cannot be empty")
 	}
+	options := req.GetParameters()
+	if value, ok := options["clusterID"]; !ok || len(value) == 0 {
+		return status.Error(codes.InvalidArgument, "Missing or empty cluster ID to provision volume from")
+	}
+	if value, ok := options["pool"]; !ok || len(value) == 0 {
+		return status.Error(codes.InvalidArgument, "Missing or empty pool name to provision volume from")
+	}
 	return nil
 }
 
-func parseVolCreateRequest(req *csi.CreateVolumeRequest) (*rbdVolume, error) {
+func (cs *ControllerServer) genVolFromCreateRequest(req *csi.CreateVolumeRequest) (*rbdVolume, error) {
```

Comment:
@ShyamsundarR may be I am missing something why the name `genVol..` , Isnt it actually do `parseVolCreateRequest' ?

Reply:
Changed back to original.